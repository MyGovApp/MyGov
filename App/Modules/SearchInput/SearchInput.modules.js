import ReduxModules from '../../Utilities/ReduxModules'
const m = new ReduxModules()

m.handleAction('SEARCH_BILLS', (s, a) => {
  return { search: a.search }
})

m.initialState = { search: '' }

export default m.createReducer()
