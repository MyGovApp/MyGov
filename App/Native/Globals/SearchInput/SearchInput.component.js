import React, { Component, PropTypes } from 'react'
import { View, Text, TextInput } from 'react-native'
import styles from './SearchInput.styles'

class SearchInput extends Component {
  render () {
    const showHide = {
      marginTop: this.props.height / 2,
      marginBottom: this.props.height / 2
    }

    if (this.props.height === 0) return null

    return (
      <View style={[ styles.searchContainer, showHide ]}>
        <Text style={styles.searchText}>Search</Text>
        <TextInput
          style={styles.input}
          clearButtonMode='always'
        />
      </View>
    )
  }
}

SearchInput.propTypes = {
  height: PropTypes.number.isRequired
}

export default SearchInput
