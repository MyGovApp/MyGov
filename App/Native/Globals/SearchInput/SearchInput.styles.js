const styles = {
  searchContainer: {
    flex: 1,
    justifyContent: 'center'
  },
  searchText: {
    textAlign: 'center',
    fontSize: 18
  },
  input: {
    borderRadius: 15,
    height: 30,
    marginRight: 10,
    marginLeft: 10,
    marginTop: 5,
    paddingLeft: 20,
    paddingRight: 20,
    backgroundColor: 'white',
    textAlign: 'center'
  }
}

export default styles
