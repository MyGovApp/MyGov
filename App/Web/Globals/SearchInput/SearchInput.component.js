import React, { Component, PropTypes } from 'react'
import s from './SearchInput.styles.scss'

class SearchInput extends Component {
  constructor () {
    super()
    this.state = {
      searchText: ''
    }
  }

  handleChange = (e) => {
    this.props.searchBills(e.target.value)
    this.setState({ searchText: e.target.value })
  }

  render () {
    return (
      <input
        type='text'
        placeholder='Search'
        value={this.state.searchText}
        onChange={this.handleChange}
        className={s.input}
      />
    )
  }
}

SearchInput.propTypes = {
}

module.exports = SearchInput
