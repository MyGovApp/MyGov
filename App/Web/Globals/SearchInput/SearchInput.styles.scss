@import "../../themes/base";

.input {
  border-radius: 13px;
  width: 100%;
  height: 26px;
  text-align: center;
  border: solid 1px $frost;
  margin: 10px 0 15px;
}

.input:focus {
  outline: none;
}
