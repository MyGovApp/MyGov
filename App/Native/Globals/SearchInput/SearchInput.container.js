import { connect } from 'react-redux'
import { searchBills } from '../../../Modules/actions'
import SearchInput from './SearchInput.component.js'

const mapDispatchToProps = {
  searchBills
}

const mapStateToProps = (state, ownProps) => ({
})

export default connect(mapStateToProps, mapDispatchToProps)(SearchInput)
