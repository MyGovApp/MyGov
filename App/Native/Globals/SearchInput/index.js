import SearchInput from './SearchInput.container.js'

export default SearchInput
