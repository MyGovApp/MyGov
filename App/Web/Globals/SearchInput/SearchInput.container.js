import { connect } from 'react-redux'
import { searchBills } from '../../../Actions/webActions'
import SearchInput from './SearchInput.component.js'

const mapDispatchToProps = {
  searchBills
}

const mapStateToProps = (state, ownProps) => ({
  search: state.SearchInput.search
})

export default connect(mapStateToProps, mapDispatchToProps)(SearchInput)
