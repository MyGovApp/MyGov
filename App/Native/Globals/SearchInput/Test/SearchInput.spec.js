import 'react-native'
import React from 'react' // eslint-disable-line
import renderer from 'react-test-renderer'
import { ComponentWrapper, ContainerWrapper } from './SearchInput.setup'

describe('Native SearchInput', () => {
  it('should render the SearchInput component correctly', () => {
    const tree = renderer.create(ComponentWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the SearchInput container correctly', () => {
    const tree = renderer.create(ContainerWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
