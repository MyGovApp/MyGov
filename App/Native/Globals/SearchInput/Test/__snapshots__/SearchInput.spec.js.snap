// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Native SearchInput should render the SearchInput component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "marginBottom": 0.5,
        "marginTop": 0.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    Search
  </Text>
  <TextInput
    clearButtonMode="always"
    onChangeText={[Function]}
    placeholder="Search"
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 30,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 5,
        "paddingLeft": 20,
        "paddingRight": 20,
        "textAlign": "center",
      }
    }
    value={undefined}
  />
</View>
`;

exports[`Native SearchInput should render the SearchInput container correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "marginBottom": 0.5,
        "marginTop": 0.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    Search
  </Text>
  <TextInput
    clearButtonMode="always"
    onChangeText={[Function]}
    placeholder="Search"
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 30,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 5,
        "paddingLeft": 20,
        "paddingRight": 20,
        "textAlign": "center",
      }
    }
    value=""
  />
</View>
`;
